lib/src/dart2_suggestors/pubspec_over_react_upgrader.dart

```diff
@@ -34,58 +37,66 @@ class PubspecOverReactUpgrader implements Suggestor {
   static final VersionRange dart2Constraint =
       VersionConstraint.parse('>=1.30.2 <3.0.0');
 
-  /// Regex that matches the dependency constraint declaration for over_react.
-  static final RegExp overReactDep = RegExp(
-    r'''^\s*over_react:\s*["']?([\d\s<>=^.]+)["']?\s*$''',
-    multiLine: true,
-  );
+  /// Constraint to update over_react to.
+  final VersionRange targetConstraint;
 
-  /// Regex that matches the `dependencies:` key in a pubspec.yaml.
-  static final RegExp dependenciesKey = RegExp(
-    r'^dependencies:$',
-    multiLine: true,
-  );
+  /// Whether or not the codemod should ignore the constraint minimum when
+  /// considering whether to write a patch.
+  final bool shouldIgnoreMin;
 
-  final VersionRange targetConstraint;
+  /// Whether or not the codemod should add `over_react` if it is not already
+  /// found.
+  final bool shouldAddDependencies;
 
-  PubspecOverReactUpgrader(this.targetConstraint);
+  PubspecOverReactUpgrader(this.targetConstraint,
+      {this.shouldAddDependencies = true})
+      : shouldIgnoreMin = false;
+
+  /// Constructor used to ignore checks and ensure that the codemod always
+  /// tries to update the constraint.
+  ///
+  /// This is useful because the codemod may want to enforce a specific
+  /// range, rather than a target upper or lower bound. The only time this
+  /// will not update the pubspec is if the target version range is equal to
+  /// the version that is already there (avoiding an empty patch error).
+  PubspecOverReactUpgrader.alwaysUpdate(this.targetConstraint,
+      {this.shouldAddDependencies = true})
+      : shouldIgnoreMin = true;
 
   @override
   Iterable<Patch> generatePatches(SourceFile sourceFile) sync* {
     final contents = sourceFile.getText(0);
-    final overReactMatch = overReactDep.firstMatch(contents);
+    final overReactMatch = overReactDependencyRegExp.firstMatch(contents);
     if (overReactMatch != null) {
       // over_react is already in pubspec.yaml
-      final line = overReactMatch.group(0);
-      final constraintValue = overReactMatch.group(1);
+      final constraintValue = overReactMatch.group(2);
       final constraint = VersionConstraint.parse(constraintValue);
-      if (constraint is VersionRange && constraint.min < targetConstraint.min) {
+
+      if (shouldUpdateVersionRange(
+          targetConstraint: targetConstraint,
+          constraint: constraint,
+          shouldIgnoreMin: shouldIgnoreMin)) {
         // Wrap the new constraint in quotes if required.
         var newValue = targetConstraint.toString();
```

Comment:
#nit
again you don't have too and its fine as is... butttt
if you wanted to use `DependencyCreator` you could remove the need for `mightNeedYamlEscaping` and just do like 

```
var replaceValue = DependencyCreator('over_react', version: targetConstraint.toString());
... 
yield Patch(
  sourceFile,
  sourceFile.span(overReactMatch.start, overReactMatch.end),
  replaceValue.toString(),
);

Reply:
I used `DependencyCreator` in every other case, but in this one I think it's simple enough as is.. and these suggestors are particularly fragile so if the current method words, I think I'll just keep is as is :)